### PR TITLE
Enforce proto ordering rules during CI lint

### PR DESCRIPTION
**Pull request by mwaskom**
We periodically (https://github.com/modal-labs/modal/pull/2875, https://github.com/modal-labs/modal-client/pull/1210, https://github.com/modal-labs/modal-client/pull/1213) spend time manually sorting our proto definitions, yet they inevitably degrade with subsequent commits.

This PR adds a CI check that enforces the following rules:

- The proto file defines types in the following order: imports, enums, messages, services
- Within each type, entities are sorted lexicographically
- Within the service entity, the RPCs for each service are also sorted

Note that this is a check-only lint; fixes will have to be manual (for now).

### CODE REVIEW

**Comment by mwaskom** [comment]
@prbot approve, this is going to attract annoying merge conflicts if it flaps in the breeze

**Comment by mwaskom** [comment]
@prbot automerge

File: tasks.py [hunk 1]
```diff
@@ -38,6 +38,52 @@ def lint(ctx, fix=False):
...
+            # This is a simplistic and sort of hacky of way of identifying the "out of order" entity,
+            # as the latter one may be the one that is misplaced. Doesn't seem worth the effort though.
+            lineno = get_first_lineno_with_prefix(proto_text, f"{a_type} {a_name}")
+            console.print(f"[bold red]Proto lint error:[/bold red] {proto_fname}:{lineno}")
+            console.print(f"\nThe {a_name} {a_type} proto is out of order relative to the {b_name} {b_type}.")
+            console.print(
+                "\nProtos should be organized into the following sections:", *sections, sep="\n - ", style="dim"
+            )
+            console.print("\nWithin sections, protos should be lexicographically sorted by name.", style="dim")
+            sys.exit(1)
```
Comment (erikbern) [inline tasks.py:73]: super minor but you might want to print _all_ issues and then exit

**Review by erikbern — COMMENTED** [review]
amazing, thanks for adding this!

**Pull request merged**